src/input/text-area.directive.ts

```diff
@@ -27,4 +27,13 @@ export class TextArea {
 	@HostBinding("class.cds--text-area--light") get isLightTheme() {
 		return this.theme === "light";
 	}
+	constructor(private elementRef: ElementRef<HTMLTextAreaElement>, private renderer: Renderer2) {}
+
+	@Input() set cols(value: number) {
+		if (value) {
+			this.renderer.setStyle(this.elementRef.nativeElement, "width", "");
+		} else {
+			this.renderer.setStyle(this.elementRef.nativeElement, "width", "100%");
+		}
+	}
```

Comment:
Not a big fan of this change for two reasons:

1. We should be avoiding inline styling as much as possible in favor of using classes. This makes the code more readable and professional.
2. I don't see any reason to include this as you can simply assign a class to a projected text area to set the width to 100%. In v4, the text area was set to 100% by default but removed in v5 to give teams more customization capability.

Other then this, I think everything makes sense to me. I agree that there should be some level of weight to the 3 states of form items. Setting the state to invalid when the other 2 states (warning/default) are enabled should set the state to invalid.

Reply:
@zvonimirfras, it would make sense that we apply this change to all form components, not just label and text area. Thoughts on this? I don't think it makes sense to release a fix not applied to all components.

Reply:
> We should be avoiding inline styling as much as possible in favor of using classes. This makes the code more readable and professional.

you are right, also this can be done in CSS with less boilerplate, and I forgot to forward the cols property to the textarea 

> I don't see any reason to include this as you can simply assign a class to a projected text area to set the width to 100%. In v4, the text area was set to 100% by default but removed in v5 to give teams more customization capability.

true, this can be easily done on the consumer's end anyway, adding this looked reasonable to me because it is the default behavior in web components and react